enlace/ajax/add_absence.php

```diff
@@ -26,11 +26,32 @@
     include "../include/dbconnopen.php";
     $user_id_sqlsafe=mysqli_real_escape_string($cnnEnlace, $_POST['user_id']);
     $date_sqlsafe=mysqli_real_escape_string($cnnEnlace, $_POST['date']);
+    $row_id_select="SELECT Absence_ID FROM Absences WHERE Participant_ID = '$user_id_sqlsafe' AND Program_Date = '$date_sqlsafe'";
 
-if ($_POST['action']=='add'){
-    $new_absence="INSERT INTO Absences (Participant_ID, Program_Date) VALUES ('".$user_id_sqlsafe."', '".$date_sqlsafe."')";
+if ($_POST['action']=='absent'){
     include "../include/dbconnopen.php";
-    mysqli_query($cnnEnlace, $new_absence);
+    $absence_id = mysqli_fetch_row(mysqli_query($cnnEnlace, $row_id_select));
+    if($absence_id) {
+        $update_absence="UPDATE Absences SET Absent = 1 WHERE Participant_ID = '$user_id_sqlsafe' AND Program_Date = '$date_sqlsafe'";
+        mysqli_query($cnnEnlace, $update_absence);
+    }
+    else{
+        $new_absence="INSERT INTO Absences (Participant_ID, Program_Date, Absent) VALUES ('".$user_id_sqlsafe."', '".$date_sqlsafe."', 1)";
+        mysqli_query($cnnEnlace, $new_absence);
+    }

```

Comment:
I tried to create absences without having run the migration, just to see what would happen, and the attempt failed silently.  It would help to add some kind of error check here, so that if the absence isn't saved the user *knows,* and the checkboxes don't just disappear when they navigate away.

Reply:
Do you mean that you want a check for validity of database, or a check that the users input has happened correctly?  Both seem out of step with the way the code currently is written (for instance, before, I believe if a failure to add an absence happened, it failed silently as well).  Don't get me wrong, I do agree with you, but I'm trying to keep budget down by deviating as little as possible from what happened before.

Reply:
Fair enough!  I'm trying to think about the best code result, but this is probably the right call for the budget.  Glad you're on it!